### DISCUSSION TITLE
Visual Studio Template

### DISCUSSION THREAD

**Discussion by russosalv** [binick/msbullet#61] [Q&A] [answered]
Ciao Nico,
if i remember well some times ago you show me a possibility to have Visual Studio Template ready.
is it still alive or i remember wrong things?

Ciao

**Comment by binick** [comment, marked as answer]
Hi Ale,

mmmh I don't remember but since I never thought before your question about the possibility of making MsBullet available through a Visual Studio Template I would say you remember wrong 😄

Anyway I have been thinking about how to make it more ready to use using an SDK for MSBuild.
I had thought about being able to make a setup within a container so that you could just compile within it but that would cause development time inconsistencies since the IDEs would not take advantage of the SDK.

To get right back to the point I see a couple of open points that need to be resolved before I can say that making a Visual Studio template.

The first is that MSBullet binds more to the repository than to a solution since MSBuild leverages the file system tree. How should the Visual Studio wizard behave if you add a uter solution in the same repo?

It might probably be possible to create a template restricted to the individual project.

If you think it would be useful for you to have a [project template](https://docs.microsoft.com/visualstudio/msbuild/how-to-use-project-sdk) let me know!

Increasing the ease of adoption of good development practices and thus MSBullet is one of the priorities 🙏

**Reply by russosalv** [reply]
Hi Nico,

thx for your answer.

Yes, i think that a vs project or maybe something simpler as startup repo to clone can be a benefit to easy adoption of MSBullet.

The idea with container should works, maybe with Docker Dev Enviroments.

I will try to create something for that, if i will success, i will let you know  :)

Ciao Grazie 👍
